Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,53 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            string EPart1=equation.Split('*')[0];
+            string EPart2=equation.Split('*')[1].Split('=')[0];
+            string EPart3=equation.Split('=')[1];
+            int result=0;
+            var answer=0.0;
+            String answer1=null, answer2=null;
+            if (EPart1.Contains("?"))

```

Comment:
Follow Single responsibility principle from SOLID principles and make your code more readable with well defined functions.

e.g you can refactor your code with below pseudo code : 


int GetMissingDigit(string multiplier1, string multiplier2, string result)
{
  // Call appropriate method as required.
  
  int multiplier1;
  if(int.TryParse(multiplier1,out multiplier1) == false)
  {
	GetMissingDigitInMultiplier(int, string, int);
  }
  elseif(int.TryParse(multiplier1,out multiplier1) == false)
  {
  
  }
  else
  {
  GetMissingDigitInResult(int,int, string);
  }
}

int GetMissingDigitInMultiplier(int multiplier1, string multiplier2, int result)
{

}

int GetMissingDigitInResult(int multiplier1, int multiplier2, string result)
{

}